Spdx license

markstur opened the pull request:
### Description

Use the short notation for SPDX-License-Identifiers in source files.   This is more consistent/accurate because you don't have to read/parse the legal language in case it was tweaked and it is more easily machine readable.  This is needed for LF.

### Checklist

#### General
- [x] I have read the appropriate contributor guide: [Python](https://github.com/i-am-bee/beeai-framework/blob/main/CONTRIBUTING.md)
/ [TypeScript](https://github.com/i-am-bee/beeai-framework/blob/main/CONTRIBUTING.md)
- [x] I have signed off on my commit: [Python instructions](https://github.com/i-am-bee/beeai-framework/blob/main/python/CONTRIBUTING.md#developer-certificate-of-origin-dco) / [TypeScript instructions](https://github.com/i-am-bee/beeai-framework/blob/main/typescript/CONTRIBUTING.md#developer-certificate-of-origin-dco)
- [x] Commit messages and PR title follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Appropriate label(s) added to PR: `Python` for Python changes, `TypeScript` for TypeScript changes

#### Code quality checks
- [x] Linting passes: Python `poe lint` or `poe lint --fix` / TypeScript `yarn lint` or `yarn lint:fix`
- [x] Formatting is applied: Python `poe format` or `poe format --fix` / TypeScript: `yarn format` or `yarn format:fix`
- [ ] Static type checks pass: Python `poe type-check`

#### Testing
- [ ] Unit tests pass: Python `poe test --type unit` / TypeScript `yarn test:unit`
- [ ] E2E tests pass: Python `poe test --type e2e` / TypeScript: `yarn test:e2e`
- [ ] Integration tests pass: Python `poe test --type integration`
- [ ] Tests are included (for bug fixes or new features)

#### Documentation
- [ ] Documentation is updated
- [ ] Embedme embeds code examples in docs. To update after edits, run: Python `poe docs --type build`

Tomas2D reviewed (changes requested):
New files were added to the project, and if I rerun the script, it does not regenerate the headers. Can you please handle it?

markstur commented:
@Tomas2D I updated the scripts to find the additional files the new files that were added and found quite a few more that were not covered due to `**` not working very well.

Not sure what is going on with the Labeler action, but I see it is failing in your other PR too.

Also since license headers in examples source looks bad in docs, changed the embedmes to skip the header.  Looks nicer.

Tomas2D commented:
I am all in regarding switching to `mise`. I just didn't have time to do it.

markstur commented:
> I am all in regarding switching to `mise`. I didn't have time to do it.

I have mixed feelings about it because of sneaky CI problems that I have been fighting, but was assuming this is the direction and started working on it yesterday.  Yesterday I had nwa issues to fix, but today I'll look into addlicense instead.

markstur commented:
> I am all in regarding switching to `mise`. I didn't have time to do it.

@Tomas2D @JanPokorny  mise and addlicense in PR #900 

My thinking was that this would merge first and I'd rebase that one, but if the other is preferred I can follow up with the doc snippet ranges separately I guess.
